### PR TITLE
kubelet starts pod after network is ready and dns service has IP assigned

### PR DESCRIPTION
**Pull request by h-w-chen**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
this PR ensures the pod's associated network is of ready state, and the dns service has IP address assigned.

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: pkg/kubelet/network/dns/dns_test.go [hunk 1]
```diff
@@ -62,6 +62,23 @@ func newTestConfigurer(recorder *record.FakeRecorder, nodeRef *v1.ObjectReferenc
 		},
 		Status: arktosv1.NetworkStatus{
 			DNSServiceIP: dnsVIPOfTestNetwork,
+			Phase:        "Ready",
+		},
+	})
+	return NewConfigurer(recorder, nodeRef, nil, clusterDNS, testClusterDNSDomain, resolverConfig, networkClient.ArktosV1())
+}
+
+func newTestConfigurerWithNotReadyNetwork(recorder *record.FakeRecorder, nodeRef *v1.ObjectReference, clusterDNS []net.IP, testClusterDNSDomain, resolverConfig string) *Configurer {
```
Comment (chenqianfzh) [inline pkg/kubelet/network/dns/dns_test.go:71]: Does it work for non-system tenant? If yes, we should have a test case for it.

File: pkg/kubelet/network/dns/dns.go [hunk 2]
```diff
@@ -84,7 +85,7 @@ func NewConfigurer(recorder record.EventRecorder, nodeRef *v1.ObjectReference, n
 	}
 }
 
-func (c *Configurer) getClusterDNS(pod *v1.Pod) []net.IP {
+func (c *Configurer) getClusterDNS(pod *v1.Pod) ([]net.IP, error) {
 	// todo: consider using exported const of default network name from arktos package
 	const defaultNetwork = "default"
 	networkName, ok := pod.Labels["arktos.futurewei.com/network"]
```
Comment (chenqianfzh) [inline pkg/kubelet/network/dns/dns.go:91]: seems we have a const for "arktos.futurewei.com/network"

File: pkg/kubelet/network/dns/dns_test.go [hunk 3]
```diff
@@ -62,6 +62,23 @@ func newTestConfigurer(recorder *record.FakeRecorder, nodeRef *v1.ObjectReferenc
 		},
 		Status: arktosv1.NetworkStatus{
 			DNSServiceIP: dnsVIPOfTestNetwork,
+			Phase:        "Ready",
```
Comment (chenqianfzh) [inline pkg/kubelet/network/dns/dns_test.go:65]: "ready" -> arktosv1.NetworkReady

File: pkg/kubelet/network/dns/dns_test.go [hunk 4]
```diff
@@ -62,6 +62,23 @@ func newTestConfigurer(recorder *record.FakeRecorder, nodeRef *v1.ObjectReferenc
...
+	networkClient := fakearktosv1.NewSimpleClientset(&arktosv1.Network{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "default",
+			Tenant: "system",
+		},
+		Spec: arktosv1.NetworkSpec{
+			Type: "test-type",
+		},
+		Status: arktosv1.NetworkStatus{
+			Phase: "Unknown",
```
Comment (chenqianfzh) [inline pkg/kubelet/network/dns/dns_test.go:81]: "unknown" -> arktosv1.NetworkUnknown

File: pkg/kubelet/network/dns/dns.go [hunk 5]
```diff
@@ -372,7 +377,11 @@ func (c *Configurer) GetPodDNS(pod *v1.Pod) (*runtimeapi.DNSConfig, error) {
 		// DNSNone should use empty DNS settings as the base.
 		dnsConfig = &runtimeapi.DNSConfig{}
 	case podDNSCluster:
-		clusterDNS := c.getClusterDNS(pod)
+		var clusterDNS []net.IP
+		if clusterDNS, err = c.getClusterDNS(pod); err != nil {
+			return nil, fmt.Errorf("failed to get DNS IP add: %v", err)
```
Comment (chenqianfzh) [inline pkg/kubelet/network/dns/dns.go:382]: add -> address

**Review by chenqianfzh — APPROVED** [review]
LGTM. Leaving some minor comments about coding and test.

File: pkg/kubelet/network/dns/dns.go (hunk 2 above)
Comment (h-w-chen) [reply pkg/kubelet/network/dns/dns.go:91]: yes; will reference the const.

File: pkg/kubelet/network/dns/dns_test.go (hunk 1 above)
Comment (h-w-chen) [reply pkg/kubelet/network/dns/dns_test.go:71]: tenancy is orthogonal to this feature. this test case seems adequate for now; if there is bug or regression, we will add necessary cases.

File: pkg/kubelet/network/dns/dns_test.go (hunk 3 above)
Comment (h-w-chen) [reply pkg/kubelet/network/dns/dns_test.go:65]: it is fine to use literal in test code for ease of code reading.

File: pkg/kubelet/network/dns/dns.go (hunk 5 above)
Comment (h-w-chen) [reply pkg/kubelet/network/dns/dns.go:382]: thanks for pointing this typo out. will update.

**Pull request merged**